authors/apps/articles/views.py

```diff
@@ -0,0 +1,57 @@
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+from .models import Article
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers
+from .serializers import ArticleSerializer, ArticleUpdateSerializer
+from .renderers import ArticleJSONRenderer
+from rest_framework import status
+from rest_framework.response import Response
+from authors.apps.utils.messages import error_messages
+from authors.apps.profiles.models import Profile
+
+
+class ArticleListCreateView(generics.ListCreateAPIView):
+    queryset = Article.objects.all()
+    serializer_class = ArticleSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleJSONRenderer,)
+
+    def perform_create(self, serializer):
+        serializer.save(
+            author=Profile.objects.get(user=self.request.user)
+        )
+
+
+class ArticleUpdateDeleteView(generics.RetrieveUpdateDestroyAPIView):
+    """ Updates and deletes an article instance """
+    serializer_class = ArticleUpdateSerializer
+    permission_class = (IsAuthenticated,)
+
+    def get_object(self):
+        slug = self.kwargs.get('slug')
+        return get_object_or_404(Article, slug=slug)
+
+    def check_if_is_author(self):
```

Comment:
@mariamiah @armstrongsouljah Already implemented this custom permission, please simply re-use his implementation